instrumentation/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava2/TracingAssembly.java

```diff
@@ -0,0 +1,318 @@
+                return new TracingConnectableFlowable(connectableFlowable, ctx);
+              }
+            }));
+
+    oldOnParallelAssembly = RxJavaPlugins.getOnParallelAssembly();
+    RxJavaPlugins.setOnParallelAssembly(
+        compose(
+            oldOnParallelAssembly,
+            new ConditionalOnCurrentContextFunction<ParallelFlowable>() {
+              @Override
+              ParallelFlowable applyActual(ParallelFlowable parallelFlowable, Context ctx) {
+                return new TracingParallelFlowable(parallelFlowable, ctx);
+              }
+            }));
+
+    enabled = true;
+  }
+
+  private static synchronized void enableCompletable() {
+    oldOnCompletableSubscribe = RxJavaPlugins.getOnCompletableSubscribe();
+    RxJavaPlugins.setOnCompletableSubscribe(
+        biCompose(
+            oldOnCompletableSubscribe,
+            new BiConditionalOnCurrentContextFunction<Completable, CompletableObserver>() {
+              @Override
+              CompletableObserver applyActual(
+                  Completable completable, CompletableObserver observer, Context ctx) {
+                try (final Scope scope = ctx.makeCurrent()) {
+                  return new TracingCompletableObserver(observer, ctx);
+                }
+              }
+            }));
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private static synchronized void enableFlowable() {
+    oldOnFlowableSubscribe = RxJavaPlugins.getOnFlowableSubscribe();
+    RxJavaPlugins.setOnFlowableSubscribe(
+        biCompose(
+            oldOnFlowableSubscribe,
+            new BiConditionalOnCurrentContextFunction<Flowable, Subscriber>() {
+
+              @Override
+              Subscriber applyActual(Flowable flowable, Subscriber subscriber, Context ctx) {
+                try (final Scope scope = ctx.makeCurrent()) {
+                  if (subscriber instanceof ConditionalSubscriber) {
+                    return new TracingConditionalSubscriber<>(
+                        (ConditionalSubscriber) subscriber, ctx);
+                  } else {
+                    return new TracingSubscriber<>(subscriber, ctx);
+                  }
+                }
+              }
+            }));
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private static synchronized void enableObservable() {
+    oldOnObservableSubscribe = RxJavaPlugins.getOnObservableSubscribe();
+    RxJavaPlugins.setOnObservableSubscribe(
+        biCompose(
+            oldOnObservableSubscribe,
+            new BiConditionalOnCurrentContextFunction<Observable, Observer>() {
+
+              @Override
+              Observer applyActual(Observable observable, Observer observer, Context ctx) {
+                try (final Scope scope = ctx.makeCurrent()) {
+                  return new TracingObserver(observer, ctx);
+                }
+              }
+            }));
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private static synchronized void enableSingle() {
```

Comment:
All those private `enable*()` methods do not need to be synchronized, right? `synchronized` in the main `enable()` method should suffice

Reply:
👍